Automattic/jetpack
# Protect: Improve manual scan flow

#### Description
This PR addresses some inconsistencies in the UI flow when processing a manual scan.

#### Changes proposed in this Pull Request:

* Adds a `START_SCAN_OPTIMISTICALLY` action to manually update the status to `scanning`
* Updates the conditional check for the in-progress display to consider the current `status`
* Updates the UI flow when processing a successful manual scan: 
   * Scan now button click
   * Display success notice
   * Dismiss success notice after 1s delay
   * Manually assign `scanning` to status 
   * Immediately display `in-progress` screen
   * Begin refreshing status continuously after 5s delay
   * Render the display accordingly

#### Other information:

#### Jetpack product discussion


- 1201069996155224-as-1203235765510738

#### Does this pull request change what data or activity we track or use?


- No

#### Testing instructions:






* Fire up a Jurassic Ninja instance w/ the Jetpack Beta plugin installed using this Protect branch
* Upgrade to Scan inclusive plan
* Queue a scan from the UI and verify that: 
   * The success notice only displays for ~1s
   * The in-progress screen displays immediately and for the duration that the scan is running
   * The new summary and threat list displays once the scan is complete and the Scan now button is no longer in a loading state
* Ensure all related actions remain functional and perform as expected
